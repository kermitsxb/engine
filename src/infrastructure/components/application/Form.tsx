import type { Props } from '@domain/engine/page/component'
import { classNames } from '../utils'

export const Form = ({
  id,
  className = '',
  Title,
  Paragraph,
  Inputs,
  Buttons,
  successMessage,
  action,
  method,
  formId,
}: Props['Form']) => (
  <section id={id} className={classNames('bg-white dark:bg-gray-900', className)} data-component="Form">
    <div className="py-8 lg:py-16 px-4 mx-auto max-w-screen-md">
      {Title && <Title className="mb-4" />}
      {Paragraph && <Paragraph className="mb-8" />}
      {successMessage ? (
        <p className="mb-8 lg:mb-16 text-center">{successMessage}</p>
      ) : (
        <div className="space-y-8">
          <form id={formId} action={action} method={method} className="space-y-8">
            {Inputs.map((Input, index) => (
              <Input key={index} />
            ))}
          </form>
          <div className="flex justify-end space-x-4">
            {Buttons.map((Button, index) => (
              <Button key={index} formId={formId} />
            ))}
          </div>
        </div>
      )}
    </div>
  </section>
)
