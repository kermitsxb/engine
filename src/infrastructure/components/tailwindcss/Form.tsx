import type { FormProps, FormResponseProps } from '/domain/entities/Form'

export const FormContainer = ({ children }: React.PropsWithChildren) => {
  return (
    <div className="w-full h-screen bg-gray-50 dark:bg-neutral-800">
      <div className="max-w-[85rem] px-4 py-10 sm:px-6 lg:px-8 lg:py-14 mx-auto">{children}</div>
    </div>
  )
}

export const FormResponse = ({ id, message }: FormResponseProps) => {
  return (
    <div id={`${id}-form-container`} className="max-w-xl mx-auto">
      <div className="text-center">
        <p className="mt-10 text-gray-600 dark:text-neutral-400">{message}</p>
      </div>
    </div>
  )
}

export const Form = ({
  id,
  title,
  description,
  submitLabel,
  children,
  formClientProps = {},
}: FormProps) => {
  return (
    <FormContainer>
      <div id={`${id}-form-container`} className="max-w-xl mx-auto">
        <div className="text-center">
          <h1 className="text-3xl font-bold text-gray-800 sm:text-4xl dark:text-white">{title}</h1>
          {description ? (
            <p className="mt-1 text-gray-600 dark:text-neutral-400">{description}</p>
          ) : null}
        </div>
        <div className="mt-12">
          <form {...formClientProps}>
            <div className="grid gap-4 lg:gap-6">
              {children}
              <div className="mt-6 grid">
                <button
                  type="submit"
                  className="w-full py-3 px-4 inline-flex justify-center items-center gap-x-2 text-sm font-medium rounded-lg border border-transparent bg-blue-600 text-white hover:bg-blue-700 focus:outline-hidden focus:bg-blue-700 disabled:opacity-50 disabled:pointer-events-none"
                >
                  {submitLabel}
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </FormContainer>
  )
}
