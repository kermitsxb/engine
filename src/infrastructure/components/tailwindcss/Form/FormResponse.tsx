import type { FormResponseProps } from '/domain/components/Form/FormResponse'

export const FormResponse = ({ id, message }: FormResponseProps) => {
  return (
    <div id={`${id}-form-container`} className="max-w-xl mx-auto">
      <div className="text-center">
        <p className="mt-10 text-gray-600 dark:text-neutral-400">{message}</p>
      </div>
    </div>
  )
}
