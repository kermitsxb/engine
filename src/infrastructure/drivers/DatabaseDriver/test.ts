import type { IDatabaseTableDriver } from '@adapter/spi/drivers/DatabaseTableSpi'
import BunTest, { afterAll } from 'bun:test'

export function testDatabaseTableDriver(
  { describe, beforeAll, it, expect }: typeof BunTest,
  setup: (date: Date) => Promise<IDatabaseTableDriver>,
  teardown?: () => Promise<void>
) {
  let databaseTable: IDatabaseTableDriver
  const date = new Date()

  beforeAll(async () => {
    databaseTable = await setup(date)
  })

  afterAll(async () => {
    if (teardown) await teardown()
  })

  describe('read', () => {
    it('should return a row with a filter', async () => {
      // WHEN
      const row = await databaseTable.read({
        field: 'name',
        operator: 'Is',
        value: 'John Doe',
      })

      // THEN
      expect(row).toStrictEqual({
        id: '1',
        fields: { name: 'John Doe' },
        created_at: date,
        updated_at: date,
      })
    })

    it('should not return a row with an empty "or" filter', async () => {
      // WHEN
      const row = await databaseTable.read({ or: [] })

      // THEN
      expect(row).toBeUndefined()
    })

    it('should not return a row with an empty "and" filter', async () => {
      // WHEN
      const row = await databaseTable.read({ and: [] })

      // THEN
      expect(row).toBeUndefined()
    })

    it('should not return a row with a filter', async () => {
      // WHEN
      const row = await databaseTable.read({
        field: 'name',
        operator: 'Is',
        value: 'Jane Doe',
      })

      // THEN
      expect(row).toBeUndefined()
    })

    it('should not return a row', async () => {
      // WHEN
      const row = await databaseTable.read({
        field: 'name',
        operator: 'Is',
        value: 'Jane Doe',
      })

      // THEN
      expect(row).toBeUndefined()
    })
  })

  describe('list', () => {
    it('should return a list of rows with no filter', async () => {
      // WHEN
      const rows = await databaseTable.list()

      // THEN
      expect(rows).toStrictEqual([
        {
          id: '1',
          fields: { name: 'John Doe' },
          created_at: date,
          updated_at: date,
        },
      ])
    })

    it('should return a list of rows with an empty "or" filter', async () => {
      // WHEN
      const rows = await databaseTable.list({ or: [] })

      // THEN
      expect(rows).toHaveLength(1)
    })

    it('should return a list of rows with an empty "and" filter', async () => {
      // WHEN
      const rows = await databaseTable.list({ and: [] })

      // THEN
      expect(rows).toHaveLength(1)
    })

    it('should return an empty list of rows with a filter', async () => {
      // WHEN
      const rows = await databaseTable.list({
        field: 'name',
        operator: 'Is',
        value: 'Jane Doe',
      })

      // THEN
      expect(rows).toHaveLength(0)
    })
  })
}
