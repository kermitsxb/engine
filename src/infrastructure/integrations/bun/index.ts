import type { Integrations } from '@adapter/spi/integrations'
import type { NotionConfig } from '@domain/integrations/Notion'
import type { PappersConfig } from '@domain/integrations/Pappers'
import type { QontoConfig } from '@domain/integrations/Qonto'
import type { NgrokConfig } from '@domain/integrations/Ngrok'

import { NotionIntegration } from '../NotionIntegration'
import { PappersBunIntegration } from './PappersBunIntegration'
import { QontoBunIntegration } from './QontoBunIntegration'
import { NgrokBunIntegration } from './NgrokBunIntegration'

export const integrations: Integrations = {
  notion: (config?: NotionConfig) => new NotionIntegration(config),
  pappers: (config?: PappersConfig) => new PappersBunIntegration(config),
  qonto: (config?: QontoConfig) => new QontoBunIntegration(config),
  ngrok: (config?: NgrokConfig) => new NgrokBunIntegration(config),
}
