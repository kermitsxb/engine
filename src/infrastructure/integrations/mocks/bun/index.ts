import type { Integrations } from '@adapter/spi/integrations'
import type { NotionConfig } from '@domain/integrations/Notion'
import type { PappersConfig } from '@domain/integrations/Pappers'
import type { QontoConfig } from '@domain/integrations/Qonto'
import type { NgrokConfig } from '@domain/integrations/Ngrok'

import { NotionBunIntegration } from './NotionIntegration'
import { PappersBunIntegration } from './PappersIntegration'
import { QontoBunIntegration } from './QontoIntegration'
import { NgrokBunIntegration } from './NgrokIntegration'

export const mocks: Integrations = {
  notion: (config?: NotionConfig) => new NotionBunIntegration(config),
  pappers: (config?: PappersConfig) => new PappersBunIntegration(config),
  qonto: (config?: QontoConfig) => new QontoBunIntegration(config),
  ngrok: (config?: NgrokConfig) => new NgrokBunIntegration(config),
}
