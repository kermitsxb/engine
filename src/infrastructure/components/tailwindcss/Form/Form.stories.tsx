import type { Meta, StoryObj } from '@storybook/react'
import { Form } from './Form'
import { SingleLineTextInput } from './Input/SingleLineTextInput'
import { SingleSelectInput } from './Input/SingleSelectInput'
import { MultipleAttachmentInput } from './Input/MultipleAttachmentInput'
import { CheckboxInput } from './Input/CheckboxInput'

const meta = {
  title: 'Form/Form',
  component: Form,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof Form>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    id: 'form',
    title: 'Form',
    description: 'This is a form',
    submitLabel: 'Submit',
    children: [
      <SingleLineTextInput field="name" label="Name" />,
      <SingleSelectInput field="status" label="Status" options={['Active', 'Inactive']} />,
      <CheckboxInput field="isActive" label="Is Active" />,
      <MultipleAttachmentInput field="attachments" label="Attachments" />,
    ],
  },
}
