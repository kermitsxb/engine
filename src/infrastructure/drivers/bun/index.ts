import type { Drivers } from '/adapter/spi/drivers'
import type { DatabaseConfig } from '/domain/services/Database'
import type { MonitorsConfig } from '/domain/services/Monitor'
import type { ServerConfig } from '/domain/services/Server'
import type { ThemeConfig } from '/domain/services/Theme'

import { DatabaseDriver } from './DatabaseDriver'
import { MonitorDriver } from './MonitorDriver'
import { ServerDriver } from './ServerDriver'
import { ThemeDriver } from './ThemeDriver'

import { drivers as common } from '../common'

export const drivers: Drivers = {
  ...common,
  monitor: (config: MonitorsConfig) => new MonitorDriver(config),
  database: (config: DatabaseConfig) => new DatabaseDriver(config),
  server: (config: ServerConfig) => new ServerDriver(config),
  theme: (config: ThemeConfig) => new ThemeDriver(config),
}
