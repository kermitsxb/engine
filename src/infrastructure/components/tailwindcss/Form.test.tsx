import { describe, it, expect } from 'bun:test'
import { Form } from './Form'
import { renderToString } from 'react-dom/server'

describe('Form', () => {
  it('should match the form snapshot', async () => {
    // GIVEN
    const form = <Form id="form" title="Form" action="/form/user" />

    // WHEN
    const html = renderToString(form)

    // THEN
    expect(html).toMatchSnapshot()
  })

  it('should match the form snapshot with description', async () => {
    // GIVEN
    const form = <Form id="form" title="Form" description="Form description" action="/form/user" />

    // WHEN
    const html = renderToString(form)

    // THEN
    expect(html).toMatchSnapshot()
  })

  it('should match the form snapshot with submit label', async () => {
    // GIVEN
    const form = <Form id="form" title="Form" submitLabel="Save" action="/form/user" />

    // WHEN
    const html = renderToString(form)

    // THEN
    expect(html).toMatchSnapshot()
  })
})
