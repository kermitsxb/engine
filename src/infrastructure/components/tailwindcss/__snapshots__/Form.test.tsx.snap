// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the form snapshot 1`] = `"<div class="w-full h-screen bg-gray-50 dark:bg-neutral-800"><div class="max-w-[85rem] px-4 py-10 sm:px-6 lg:px-8 lg:py-14 mx-auto"><div class="max-w-xl mx-auto"><div class="text-center"><h1 class="text-3xl font-bold text-gray-800 sm:text-4xl dark:text-white">Form</h1></div><div class="mt-12"><form action="/form/user" method="POST"><div class="grid gap-4 lg:gap-6"><div class="mt-6 grid"><button type="submit" class="w-full py-3 px-4 inline-flex justify-center items-center gap-x-2 text-sm font-medium rounded-lg border border-transparent bg-blue-600 text-white hover:bg-blue-700 focus:outline-hidden focus:bg-blue-700 disabled:opacity-50 disabled:pointer-events-none"></button></div></div></form></div></div></div></div>"`;

exports[`Form should match the form snapshot with description 1`] = `"<div class="w-full h-screen bg-gray-50 dark:bg-neutral-800"><div class="max-w-[85rem] px-4 py-10 sm:px-6 lg:px-8 lg:py-14 mx-auto"><div class="max-w-xl mx-auto"><div class="text-center"><h1 class="text-3xl font-bold text-gray-800 sm:text-4xl dark:text-white">Form</h1><p class="mt-1 text-gray-600 dark:text-neutral-400">Form description</p></div><div class="mt-12"><form action="/form/user" method="POST"><div class="grid gap-4 lg:gap-6"><div class="mt-6 grid"><button type="submit" class="w-full py-3 px-4 inline-flex justify-center items-center gap-x-2 text-sm font-medium rounded-lg border border-transparent bg-blue-600 text-white hover:bg-blue-700 focus:outline-hidden focus:bg-blue-700 disabled:opacity-50 disabled:pointer-events-none"></button></div></div></form></div></div></div></div>"`;

exports[`Form should match the form snapshot with submit label 1`] = `"<div class="w-full h-screen bg-gray-50 dark:bg-neutral-800"><div class="max-w-[85rem] px-4 py-10 sm:px-6 lg:px-8 lg:py-14 mx-auto"><div class="max-w-xl mx-auto"><div class="text-center"><h1 class="text-3xl font-bold text-gray-800 sm:text-4xl dark:text-white">Form</h1></div><div class="mt-12"><form action="/form/user" method="POST"><div class="grid gap-4 lg:gap-6"><div class="mt-6 grid"><button type="submit" class="w-full py-3 px-4 inline-flex justify-center items-center gap-x-2 text-sm font-medium rounded-lg border border-transparent bg-blue-600 text-white hover:bg-blue-700 focus:outline-hidden focus:bg-blue-700 disabled:opacity-50 disabled:pointer-events-none">Save</button></div></div></form></div></div></div></div>"`;
